raiden/network/rpc/client.py

```diff
@@ -221,13 +222,20 @@ def __init__(
         self.gasprice = cache_wrapper(self._gasprice)
 
         # web3
-        self.web3: Web3 = Web3(HTTPProvider(endpoint))
-        # we use a PoA chain for smoketest, use this middleware to fix this
-        self.web3.middleware_stack.inject(geth_poa_middleware, layer=0)
-
-        # create the connection test middleware
-        connection_test = make_connection_test_middleware(self)
-        self.web3.middleware_stack.inject(connection_test, layer=0)
+        if web3 is None:
+            self.web3: Web3 = Web3(HTTPProvider(endpoint))
+        else:
+            self.web3 = web3
+        try:
```

Comment:
why is this `try` clause necessary

Reply:
Web3 raises a value error if you are trying to inject middleware that already exists (which is what happens with `tester` - `web3` fixture is session scoped)